qemu/tests/qemu_guest_agent_snapshot.py

```diff
@@ -55,18 +55,24 @@ def _action_after_fsthaw(self, *args):
         if self.bg:
             self.bg.join()
         # Make sure the returned file is identical to the original one
-        self.host_path_returned = "%s-returned" % self.host_path
-        self.vm.copy_files_from(self.guest_path, self.host_path_returned)
-        error_context.context("comparing hashes", logging.info)
-        self.curr_hash = crypto.hash_file(self.host_path_returned)
-        if self.orig_hash != self.curr_hash:
-            self.test.fail("Current file hash (%s) differs from "
-                           "original one (%s)" % (self.curr_hash,
-                                                  self.orig_hash))
-
-        error_context.context("Reboot and shutdown guest.")
-        self.vm.reboot()
-        self.vm.destroy()
+        try:
+            self.host_path_returned = "%s-returned" % self.host_path
+            self.vm.copy_files_from(self.guest_path, self.host_path_returned)
+            error_context.context("comparing hashes", logging.info)
+            self.curr_hash = crypto.hash_file(self.host_path_returned)
+            if self.orig_hash != self.curr_hash:
```

Comment:
@xiagao line 34,  can bg.start() make sure hash value is correct after transferring data from host to guest?  If the first time is wrong, then this time  must orig_hash != self.curr_hash.

Reply:
@xiangchunfu If the file is changed during transfer, it's a product bug.
The steps is:
1) create a file in host. -original file in host
2) while transfering data do fsfreeze operation.
3) after thaw and data transfer finished, copy the file from guest to host. -the current file.
4) compare the two files hash value in host.

Reply:
For function testing,  looks good to me. 
Acked-by: xfu@redhat.com